### DISCUSSION TITLE
Migrating from old Docker version to new

### DISCUSSION THREAD

**Discussion by lnlyssg** [pixelfed/pixelfed#4977] [Support]
Following the recent release I'm trying to follow the [Path A migration](https://jippi.github.io/pixelfed-docs-next/pr-preview/pr-1/running-pixelfed/docker/migrate/v2.0.html#a-anonymous-volumes) but I can't get it working:

```
❯ docker compose -f docker-compose.migrate.yml run migrate bash
WARN[0000] The "DOCKER_APP_HOST_STORAGE_PATH" variable is not set. Defaulting to a blank string.
WARN[0000] The "DOCKER_DB_HOST_DATA_PATH" variable is not set. Defaulting to a blank string.
WARN[0000] The "DOCKER_REDIS_HOST_DATA_PATH" variable is not set. Defaulting to a blank string.
invalid spec: :/migrate/app-storage/new: empty section between colons
```
I'm guessing I've missed something but I'm not sure what.... I can see these missing values are in my env.docker file but do they need adding to the docker-compose.migrate.yml file?

**Comment by jippi** [comment]
Are you on a recent docker compose version? (not one from Ubuntu or Debian directly - but from Docker.com own repository)?

**Reply by lnlyssg** [reply]
It's the one from Docker and I'm on v2.24.5 currently

**Reply by jippi** [reply]
is your current `.env` based on `.docker.env`?

**Reply by lnlyssg** [reply]
Aha, I see the issue - the step in the docs to rename env.new to .env is _after_ the migration steps (this probably needs to be moved earlier), my file was still named .env.new so I have now renamed it to .env and have been able to get slightly further but now have a new error:

```
❯ docker compose -f docker-compose.migrate.yml run migrate bash
[+] Creating 1/0
WARN[0000] Found orphan containers ([pixelfed-app-1 pixelfed-worker-1 pixelfed-redis-1 pixelfed-db-1]) for this project. If you removed or rename[+] Creating 1/1n your compose file, you can run this command with the --remove-orphans flag to clean it up.
 ✔ Network pixelfed_default  Created                                                                                                        0.1s
[+] Running 7/7
 ✔ migrate 6 layers [⣿⣿⣿⣿⣿⣿]      0B/0B      Pulled                                                                                         2.3s
   ✔ c9b1b535fdd9 Pull complete                                                                                                             0.3s
   ✔ 19758234142e Pull complete                                                                                                             0.5s
   ✔ 8505412ea0bd Pull complete                                                                                                             0.4s
   ✔ f7a476971866 Pull complete                                                                                                             0.9s
   ✔ 8cdb1f256e11 Pull complete                                                                                                             1.1s
   ✔ 35cb69246ea7 Pull complete                                                                                                             0.9s
exec /bin/bash: exec format error
```

**Reply by jippi** [reply]
What CPU architecture /device are you running it on? it looks like the wrong CPU architecture is being used causing it to be unable to execute 🤔

**Reply by lnlyssg** [reply]
This is an Arm machine

**Reply by jippi** [reply]
Ah, that might be the issue - I did not check the migration image for `arm` support

Can you try to change the `image:` to `secoresearch/rsync` in the migrate file? :)

**Reply by lnlyssg** [reply]
It's already set to that... I just tried changing to servercontainers/rsync which seems to be working OK so far

**Reply by jippi** [reply]
ooops, that was what I meant - bad copy paste :) 

Okay, so the rsync container need arm support, got it, can you please file a ticket for that? ❤️

**Reply by lnlyssg** [reply]
Will do. One more issue though!

The worker is now crash looping. I had OAUTH enabled on my old .env so set it to true in the new .env but I think this may be causing an issue from the output.... (I've removed my site from the container name and replaced with xxx)

```
photos.xxx-worker  | [entrypoint / 11-first-time-setup.sh] - 👷 Running [php artisan passport:keys] as [www-data]
photos.xxx-worker  | [entrypoint / 11-first-time-setup.sh] - (stdout) Encryption keys already exist. Use the --force option to overwrite them.
photos.xxx-worker  | [entrypoint / 11-first-time-setup.sh] - ERROR - ❌ Error!
photos.xxx-worker  | [entrypoint / entrypoint.sh] - 🔓 Releasing lock [/var/www/storage/docker/lock/entrypoint.sh]
```

**Reply by jippi** [reply]
The `info` bug here should cover it: https://jippi.github.io/pixelfed-docs-next/pr-preview/pr-1/running-pixelfed/docker/migrate/v2.0.html#_2-migrate-env-file

you need to disable the run of any first-time-setup steps when migrating

**Reply by lnlyssg** [reply]
Thanks I had missed that. One last query (I hope):

My db output is filled with messages like this but I don't know why

```
photos.xxx-db  | 2024-03-05 21:03:30 3 [Warning] Access denied for user 'mysql'@'127.0.0.1' (using password: NO)
photos.xxx-db  | 2024-03-05 21:03:30 4 [Warning] Access denied for user 'mysql'@'localhost' (using password: NO)
```

**Reply by jippi** [reply]
If everything else is working, then it's likely the "docker health check" system 🤔 

The Docker setup assumes the `MARIADB_ROOT_PASSWORD` is the same as your `DB_PASSWORD` which might not be true on existing systems

**Reply by lnlyssg** [reply]
Thanks, everything else is now working Ok so it must be the health check.

**Reply by jippi** [reply]
Filed https://github.com/pixelfed/pixelfed/issues/4979 and https://github.com/pixelfed/pixelfed/issues/4980 so I don't forget  - thank you :)
